manifests/mod/jk.pp

```diff
@@ -0,0 +1,105 @@
+# Class apache::mod::jk
+#
+# Manages mod_jk connector
+#
+# All parameters are optional. When undefined, some receive default values,
+# while others cause an optional directive to be absent
+#
+# For help on parameters, pls see official reference at:
+# https://tomcat.apache.org/connectors-doc/reference/apache.html
+#
+class apache::mod::jk (
+  $workers_file          = undef,
+  $worker_property       = {},
+  $shm_file              = "${::apache::logroot}/jk-runtime-status",
```

Comment:
in the case of these logroot dependencies, you could just set these to the file's name and then prepend the logroot path below. this would be a more predictable pattern for those who are wanting to override anyway.